### PR TITLE
feature "implement more understanble builder ... moreThan"

### PR DESCRIPTION
**Pull request by lolo8304**
implement nicer versions of builder pattern for count.
Testcases added and adapted / API broken, no deprecation

### CODE REVIEW

File: corda-common/src/main/java/ch/cordalo/corda/common/contracts/StateVerifier.java [hunk 1]
```diff
@@ -177,23 +177,23 @@ public StateVerifier notEmpty(String text) {
...
-    }
     public StateVerifier count(int size) {
         return new Count(this, size).verify();
     }
-    public StateVerifier max(int size) {
+    public StateVerifier counMax(int size) {
         return new Max(this, size).verify();
     }
-    public StateVerifier min(int size) {
+    public StateVerifier counMin(int size) {
```
Comment (cedricwalter) [inline corda-common/src/main/java/ch/cordalo/corda/common/contracts/StateVerifier.java]: tippfehler it it not countMin()

File: corda-common/src/main/java/ch/cordalo/corda/common/contracts/StateVerifier.java [hunk 2]
```diff
@@ -618,25 +618,42 @@ protected StateVerifier verify() {
...
         @Override
         protected StateVerifier verify() {
-            super.verify();
             requireThat(req -> {
-                req.using(s("List must more than 1 entry."), this.getParentList().size() > 1);
+                if (this.size() == 1) {
+                    req.using(s("List must have less than " + this.size() + " entry."), this.getParentList().size() < this.size());
+                } else {
+                    req.using(s("List must have less than " + this.size() + " entries."), this.getParentList().size() < this.size());
+                }
```
Comment (cedricwalter) [inline corda-common/src/main/java/ch/cordalo/corda/common/contracts/StateVerifier.java]: use instead (no if, less code)
`req.using(s("List must have less than " + this.size() + " entrie(s)."), this.getParentList().size() < this.size());`

File: corda-common/src/main/java/ch/cordalo/corda/common/contracts/StateVerifier.java [hunk 3]
```diff
@@ -177,23 +177,23 @@ public StateVerifier notEmpty(String text) {
...
         return new MoreThanZero(this).verify();
     }
-    public StateVerifier moreThanZero(int size) {
-        return new MoreThanZero(this, size).verify();
-    }
     public StateVerifier count(int size) {
         return new Count(this, size).verify();
     }
-    public StateVerifier max(int size) {
+    public StateVerifier counMax(int size) {
```
Comment (cedricwalter) [inline corda-common/src/main/java/ch/cordalo/corda/common/contracts/StateVerifier.java]: input().hasMaxCount(100) 
input().maxCount(100) 
is not better?

**Review by cedricwalter — CHANGES_REQUESTED** [review]

File: corda-common/src/main/java/ch/cordalo/corda/common/contracts/StateVerifier.java (hunk 1 above)
Comment (cedricwalter) [reply corda-common/src/main/java/ch/cordalo/corda/common/contracts/StateVerifier.java]: good idea, it done

Comment (lolo8304) [reply corda-common/src/main/java/ch/cordalo/corda/common/contracts/StateVerifier.java]: good idea

**Review by cedricwalter — APPROVED** [review]

**Pull request merged**